pkg/cmd/provisioning/gcp/create_workload_identity_pool.go

```diff
@@ -94,20 +96,36 @@ func createWorkloadIdentityPool(ctx context.Context, client gcp.Client, name, pr
 			return errors.Wrap(err, fmt.Sprintf("Failed to save script to create workload identity pool %s locally at %s", name, createIdentityPoolScriptFullPath))
 		}
 	} else {
-		pool := &iam.WorkloadIdentityPool{
-			Name:        name,
-			DisplayName: name,
-			Description: createdByCcoctl,
-			State:       "ACTIVE",
-			Disabled:    false,
-		}
-
 		parentResourceForPool := fmt.Sprintf("projects/%s/locations/global", project)
-		_, err := client.CreateWorkloadIdentityPool(ctx, parentResourceForPool, name, pool)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create workload identity pool %s", name)
+		poolResource := fmt.Sprintf("%s/workloadIdentityPools/%s", parentResourceForPool, name)
+		resp, err := client.GetWorkloadIdentityPool(ctx, poolResource)
+		if resp != nil && resp.State == "DELETED" {
+			log.Printf("Workload identity pool %s was deleted, undeleting it", name)
+			_, err := client.UndeleteWorkloadIdentityPool(ctx, poolResource, &iam.UndeleteWorkloadIdentityPoolRequest{})
+			if err != nil {
+				return errors.Wrapf(err, "failed to undelete workload identity pool %s", name)
+			}
+		} else if err != nil {
+			if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 && strings.Contains(gerr.Message, "Requested entity was not found") {
+				pool := &iam.WorkloadIdentityPool{
+					Name:        name,
+					DisplayName: name,
+					Description: createdByCcoctl,
+					State:       "ACTIVE",
+					Disabled:    false,
+				}
+
+				_, err := client.CreateWorkloadIdentityPool(ctx, parentResourceForPool, name, pool)
+				if err != nil {
+					return errors.Wrapf(err, "failed to create workload identity pool %s", name)
+				}
+				log.Printf("Workload identity pool created with name %s", name)
+			} else {
+				return errors.Wrapf(err, "failed to check if there is existing workload identity pool %s", name)
+			}
+		} else {
+			log.Printf("Workload identity pool %s already exists and is owned by the user", name)
```

Comment:
How do we know that this identity pool is really owned by the user?

Reply:
I think this statement is wrong. This should be `pool is owned by the project` instead.

Reply:
Let me just reframe it to `log.Printf("Workload identity pool %s already exists", name)` to avoid confusion